src/main/java/com/softserve/academy/spaced/repetition/controller/CourseRatingController.java

```diff
@@ -25,21 +25,20 @@
     @Autowired
     private CourseRatingService courseRatingService;
 
-    @GetMapping("api/rate/course/{id}")
-    public ResponseEntity<CourseRatingPublicDTO> getCourseRatingById(@PathVariable Long id) {
+    @GetMapping("api/course/{courseId}/rating/{id}")
+    public ResponseEntity<CourseRatingPublicDTO> getCourseRatingById(@PathVariable Long courseId, @PathVariable Long id) {
         CourseRating courseRating = courseRatingService.getCourseRatingById(id);
-        Link selfLink = linkTo(methodOn(CourseRatingController.class).getCourseRatingById(courseRating.getId())).withRel("courseRating");
+        Link selfLink = linkTo(methodOn(CourseRatingController.class).getCourseRatingById(courseRating.getCourse().getId(), courseRating.getId())).withRel("courseRating");
         CourseRatingPublicDTO courseRatingDTO = DTOBuilder.buildDtoForEntity(courseRating, CourseRatingPublicDTO.class, selfLink);
         return new ResponseEntity<>(courseRatingDTO, HttpStatus.OK);
     }
 
     @PostMapping("/api/private/course/{courseId}")
-    public ResponseEntity<DTO<CourseRating>> addCourseRating(@RequestBody CourseRating courseRating, @PathVariable Long courseId) throws RatingsBadValueException, NotAuthorisedUserException {
-        if ((courseRating.getRating() >= MIN_RATING) && (courseRating.getRating() <= MAX_RATING)) {
+    public ResponseEntity addCourseRating(@RequestBody String rating, @PathVariable Long courseId) throws RatingsBadValueException, NotAuthorisedUserException {
```

Comment:
Why rating is of String type?

Reply:
@antonbakalets, because when it was of int or Integer type, Spring couldn't parse this value.

Reply:
Ok, but I can't leave it as is. This way you can get a unhandled NumberFormatException.

Even a dto with one integer field would be a better solution. This way conversion and exception handling will be Spring responsibility.